sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/GuardTests.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Messaging.EventHubs.Core;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   The suite of tests for the <see cref="Guard" />
+    ///   class.
+    /// </summary>
+    ///
+    [TestFixture]
+    [TestCategory(Category.BuildVerification)]
+    public class GuardTests
+    {
+        /// <summary>
+        ///   Provides the invalid test cases for the <see cref="Guard.ArgumentNotNegative" /> tests.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ArgumentNotNegativeForTimeSpanInvalidCases()
+        {
+            yield return new object[] { TimeSpan.FromMilliseconds(-1) };
+            yield return new object[] { TimeSpan.FromSeconds(-2) };
+            yield return new object[] { TimeSpan.FromHours(-3) };
+            yield return new object[] { TimeSpan.FromDays(-0.3) };
+            yield return new object[] { TimeSpan.FromTicks(-100) };
+        }
+
+        /// <summary>
+        ///   Provides the valid test cases for the <see cref="Guard.ArgumentNotNullOrEmpty" /> tests.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ArgumentNotNegativeForTimeSpanValidCases()
+        {
+            yield return new object[] { TimeSpan.FromMilliseconds(0) };
+            yield return new object[] { TimeSpan.FromSeconds(1) };
+            yield return new object[] { TimeSpan.FromHours(1) };
+            yield return new object[] { TimeSpan.FromDays(0.3) };
+            yield return new object[] { TimeSpan.FromTicks(1) };
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="Guard.ArgumentNotNull" /> method.
+        /// </summary>
+        ///
+        [Test]
+        public void ArgumentNotNullEnforcesInvariants()
+        {
+            Assert.Throws<ArgumentNullException>( () => Guard.ArgumentNotNull("argument", null));
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="Guard.ArgumentNotNull" /> method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase("")]
+        [TestCase("  ")]
+        [TestCase(3)]
+        [TestCase(typeof(Guard))]
+        [TestCase(ConnectionType.AmqpTcp)]
+        [TestCase(new[] { 1, 3, 4 })]
+        public void ArgumentNotNullAllowsValidValues(object value)
+        {
+            Assert.That(() => Guard.ArgumentNotNull(nameof(value), value), Throws.Nothing);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="Guard.ArgumentNotNullOrEmpty" /> method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ArgumentNotNullOrEmptyEnforcesInvariants(string value)
+        {
+            Assert.That(() => Guard.ArgumentNotNullOrEmpty(nameof(value), value), Throws.InstanceOf<ArgumentException>());
```

Comment:
Is there a reason you sometimes use `Assert.Throws` and sometimes `Throws.InstanceOf<ArgumentException>`?

Reply:
`Throws.None` is the form for asserting that no exception occurs.  `Throws.InstanceOf<T>` is asserting that an exception of a specific type is thrown.  This is a weird class in that it doesn't so much do anything if there isn't a violation of invariants so, the only thing that can really be checked is "hey, did this thing throw?"

Reply:
I'm fine with Throws.None I was asking about the following:

``` C#
Assert.Throws<ArgumentNullException>( () => Guard.ArgumentNotNull("argument", null));

// vs 

Assert.That(() => Guard.ArgumentNotNullOrEmpty(nameof(value), value), Throws.InstanceOf<ArgumentException>());
```

Reply:
oh... yeah.  That's a mistake on my part, left from the xUnit changeover.  I'll fix that up.